src/User/components/Card/CardShop.jsx

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getProductById } from "../../../redux/thunk/product.thunk";
+import CardItem from "../CardItem/CardItem";
+import "./CardShop.scss";
+
+function CardShop() {
+  const dispatch = useDispatch();
+
+  const { filterProduct } = useSelector((state) => state.products);
+
+  const [id, setId] = useState("");
+
+  useEffect(() => dispatch(getProductById(id)), [id]);
+
+  return filterProduct.map((products, key) => {
+    return (
+      <>
+        <CardItem index={key} product={products} lg='4' />
+      </>
+    );
+  });
```

Comment:
```suggestion
  return filterProduct.map((product, key) => (
        <CardItem key={product.id} product={product} lg='4' />
    )
  );
```
Nên bổ sung key bằng ID nếu có thể thay vì index
Lí do xem tại đây nhé
https://viblo.asia/p/su-quan-trong-cua-key-props-trong-reactjs-jvElaAXolkw